vizro-core/src/vizro/models/_components/graph.py

```diff
@@ -71,6 +75,39 @@ def __getitem__(self, arg_name: str):
             return self.type
         return self.figure[arg_name]
 
+    # Interaction methods
+    def _get_figure_interaction_input(self) -> Dict[str, State]:
+        """Required properties when using pre-defined `filter_interaction`."""
+        return {
+            "clickData": State(component_id=self.id, component_property="clickData"),
+            "modelID": State(component_id=self.id, component_property="id"),  # required, to determine triggered model
+        }
+
+    def _filter_interaction(
+        self, data_frame: pd.DataFrame, target: str, ctd_filter_interaction: Dict[str, CallbackTriggerDict]
+    ) -> pd.DataFrame:
+        """Function to be carried out for pre-defined `filter_interaction`."""
+        ctd_click_data = ctd_filter_interaction["clickData"]
+        if not ctd_click_data["value"]:
+            return data_frame
+
+        source_graph_id: ModelID = ctd_click_data["id"]
+        source_graph_actions = _get_component_actions(model_manager[source_graph_id])
+        try:
+            custom_data_columns = model_manager[source_graph_id]["custom_data"]
+        except KeyError as exc:
+            raise KeyError(f"No `custom_data` argument found for source graph with id {source_graph_id}.") from exc
+
+        customdata = ctd_click_data["value"]["points"][0]["customdata"]
+
+        for action in source_graph_actions:
+            if action.function._function.__name__ != "filter_interaction" or target not in action.function["targets"]:
+                continue
+            for custom_data_idx, column in enumerate(custom_data_columns):
+                data_frame = data_frame[data_frame[column].isin([customdata[custom_data_idx]])]
+
+        return data_frame
+
```

Comment:
It looks we decided to go with the "Inserting 'actions info' and dash dependencies as a method of the model." approach. I would really like if you can update me, what happened with other approaches we were considering, and why we finally decided to go with this one.

Reply:
As mentioned briefly on a call:
- having different models is final for at least this case
- placing the actions info here is not, although it could be our final approach at least for the case of Callables inserted into models